backend/transcription.py

```
@@ -35,35 +42,94 @@ def get_groq_client():
+    try:
+        # Groq's API expects a file tuple: (filename, file_bytes)
+        transcription = client.audio.transcriptions.create(
+            file=(filename, audio_data),
+            model="whisper-large-v3-turbo",
+            temperature=0,
+            language=language,
+            response_format="verbose_json",
+        )
+
+        logger.info(f"Transcribed audio: {len(audio_data)} bytes -> {len(transcription.text)} chars")
+        return transcription.text
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)